stream.c

```diff
@@ -497,11 +497,11 @@ static void* handle_md5_digest(void* param)
 
     strncpy(server_user, p->conf->stream_authentication, h-p->conf->stream_authentication);
     server_user[h - p->conf->stream_authentication] = '\0';
-    strncpy(server_pass, h + 1, strlen(h + 1));
+    memcpy(server_pass, h + 1, strlen(h + 1));
```

Comment:
This should stay as `strncpy`. The problem here is that `server_pass` was malloc'd in line 490 as `strlen(h) + 1` size. So updating the third argument gives:

```
strncpy(server_pass, h + 1, strlen(h) + 1);
```

It may be better instead however to save the `server_pass` size back in line 490 and pass it as the third argument here so that it's more obvious what's happening. That would also be better in case `h` is ever modified later.

Reply:
done.